src/validation.cpp

```diff
@@ -2508,22 +2512,23 @@ bool CChainState::DisconnectTip(BlockValidationState& state, const CChainParams&
     if (!FlushStateToDisk(chainparams, state, FlushStateMode::IF_NEEDED))
         return false;
 
-    if (disconnectpool) {
+    if (disconnectpool && m_mempool) {
+        AssertLockHeld(m_mempool->cs);
```

Comment:
Isn't the asserion redundant with [line 2491](https://github.com/bitcoin/bitcoin/pull/19629/commits/faab637301a336c318ea8ee0187a74a2c7a92ef3#diff-24efdb00bfbe56b140fb006b562cc70bR2491) ?

Reply:
Locally it wouldn't compile for me with clang as soon as I remove it